strings/base_coroutine_threadpool.h

```diff
@@ -39,14 +39,39 @@ namespace winrt::impl
         return 0;
     };
 
+    inline auto resume_context(com_ptr<IContextCallback> const& context, std::experimental::coroutine_handle<> handle)
+    {
+        com_callback_args args{};
+        args.data = handle.address();
+
+        check_hresult(context->ContextCallback(resume_apartment_callback, &args, guid_of<ICallbackWithNoReentrancyToApplicationSTA>(), 5, nullptr));
+    }
+
     inline auto resume_apartment(com_ptr<IContextCallback> const& context, std::experimental::coroutine_handle<> handle)
     {
         if (context)
         {
-            com_callback_args args{};
-            args.data = handle.address();
-
-            check_hresult(context->ContextCallback(resume_apartment_callback, &args, guid_of<ICallbackWithNoReentrancyToApplicationSTA>(), 5, nullptr));
+            if (requires_apartment_context())
+            {
+                struct params
+                {
+                    com_ptr<IContextCallback> context;
+                    std::experimental::coroutine_handle<> handle;
+                };
+                params p{ context, handle };
```

Comment:
the lifetime of this needs to be managed, passing a shared_ptr or transferring ownership of a unique_ptr, as I don't see anything that forces the thread function to be synchronized with this.